### ISSUE TITLE
Assert "isCloseTo(double expected, Percentage p)" doesn't work for negative expected values

### ISSUE THREAD

**Issue by tomyy** [joel-costigliola/assertj-core#611]
I have two tests whether an actual value is within a given percentage of the expected value.

The test for a positive expected value works without problems.
The test for a negative expected value throws `IllegalArgumentException`

My test methods are:
```
@Test
public void checkPositiveNumber() {
    assertThat(11d).isCloseTo(10d, withinPercentage(10));
}

@Test
public void checkNegativeNumber() {
    assertThat(-11d).isCloseTo(-10d, withinPercentage(10));
}
```
I've tested only for doubles but I suspect the same problem exists for all numeric types.

**Comment by joel-costigliola** [comment]
True :crying_cat_face: 
We screwed up, the computed offset is negative, we should have used an absolute values.

Thanks for reporting this. will be fixed in  2.4.0/3.4.0

**Comment by joel-costigliola** [comment]
Thanks @PascalSchumacher I have integrated your PR and took the opportunity to refactor  some duplicated code in "numbers" classes.
